_pages/calculator.html

```diff
@@ -171,50 +177,62 @@ <h3>Calculator</h3>
 const targetInputs = lockButtons.replace(/lock/g, "tar");
 function handleLock() {
     // Reset everything to normal
-    $(lockButtons).removeAttr('disabled');
-    $(targetInputs).removeAttr('disabled');
-    $(lockButtons).text("calculate me!");
+    document.querySelectorAll('[data-cw-lock]').forEach(el => {
+        el.disabled  = false
+        el.innerText = 'calculate me!'
+    })
+    document.querySelectorAll('[data-cw-tar]').forEach(el => el.disabled = false)
+
 
-    const input = $("#" + this.id.replace(/lock/g, "tar"));
-    input.val("");
-    input.attr("disabled", true);
+    const input = document.querySelector(`[data-cw-tar="${this.dataset.cwLock}"]`)
+    input.disabled = true
+    input.value    = ''
 
-    $(this).attr("disabled", true);
-    $(this).text('calculated');
+    this.disabled = true
+    this.innerText = 'calculated!'
 
     updateCalculation();
 }
 
 function changeCourse() {
-    const acronym = $(this).find("option:selected").data('acronym');
+    const acronym = document.querySelector('#course-list').value
+
     const course = courseData[acronym];
+    const courseEl = document.querySelector('#course-descriptor')
+    console.log('changed course')
     if (!course) {
-        $("#course-descriptor").hide();
+        courseEl.style.display = 'none'
+
         return;
     }
 
-    $("#course-descriptor").show();
-    $("#course-descriptor").attr('href', course.euclid_url);
+    courseEl.style.display = 'block'
+    courseEl.setAttribute('href', course.euclid_url)
 
     if (course.cw_exam_ratio === null) {
         showCalculatorError("course.inf.ed.ac.uk is missing a cw/exam ratio for this course");
         return;
     }
 
-    $("#cw-w8-cw").val(course.cw_exam_ratio[0]);
-    $("#cw-w8-exam").val(course.cw_exam_ratio[1]);
+    getEl('w8', 'cw').value = course.cw_exam_ratio[0]
+    getEl('w8', 'exam').value = course.cw_exam_ratio[1]
 
     updateCalculation();
 }
 
 function loadCourseCalculator() {
-    $("#course-list").change(changeCourse)
+    document.querySelector('#course-list').addEventListener('change', changeCourse)
+
+    document.querySelectorAll('[data-cw-tar]').forEach(el => el.addEventListener('input', updateCalculation))
+    document.querySelectorAll('[data-cw-w8]' ).forEach(el => el.addEventListener('input', updateCalculation))
+    document.querySelectorAll('[data-cw-lock]').forEach(el => el.addEventListener('click', handleLock))
 
-    $("input").on('input', updateCalculation);
-    $(lockButtons).click(handleLock);
 
-    handleLock.bind($("#cw-lock-all")[0])();
+    // fire handler to initially populate the commits
+    try {
+        handleLock()
```

Comment:
This no longer pre-locks the "overall" field, causing an error if you select a course before clicking a "calculate me" button.

The `.bind()` in the old code would set `this` inside `handleLock`, and this is no longer being applied. If `bind()` is brought back, `handleLock` should work correctly again, and you wouldn't need the try/catch.